File: CarCity.Domain/Entities/Car.cs
namespace CarCity.Domain.Entities
{
    public class Car : BaseEntity
    {
        public string Brand { get; set; }
        public string Model { get; set; }
        public string Year { get; set; }
        public string KM { get; set; }
        public string Color { get; set; }
        public string ImageUrl { get; set; }
        public int CarTypeId { get; set; }
        public CarType CarType { get; set; }
    }
}

File: NextJSAppCarGallery/models/car.ts
import { CarType } from "./carType";

export interface Car {
    id: number;
    brand: string,
    model: any,
    year: Date,
    km: string,
    color: string,
    imageUrl: string,
    carType: CarType,
    createdDate: Date,
}

export interface CarFormValues {
    brand: string;
    model: string;
    year: string;
    km: string;
    color: string;
    imageUrl: string;
    carTypeId: number;
    // constructor(car?: CarFormValues) {
    //     if (car) {
    //         this.brand = car?.brand
    //         this.model = car?.model
    //         this.year = car?.year
    //         this.km = car?.km
    //         this.color = car?.color
    //         this.imageUrl = car?.imageUrl
    //         this.carTypeId = car?.carTypeId
    //     }
    // }
}